### PR TITLE
Test migration to openstack afterburn provider

### PR DESCRIPTION
**Pull request by alejandro-ripoll**
# Test migration to openstack afterburn provider

Test migration to `openstack` afterburn provider. 
Closes: https://github.com/flatcar/Flatcar/issues/1445

## How to use

[ describe what reviewers need to do in order to validate this PR ]

## Testing done

CI: http://jenkins.infra.kinvolk.io:8080/job/container/job/packages_all_arches/4058/cldsv/

- [ ] Changelog entries added in the respective `changelog/` directory (user-facing change, bug fix, security fix, update)
- [ ] Inspected CI output for image differences: `/boot` and `/usr` size, packages, list files for any missing binaries, kernel modules, config files, kernel modules, etc.

### CODE REVIEW

File: sdk_container/src/third_party/coreos-overlay/sys-kernel/bootengine/bootengine-9999.ebuild
[the remark this answers is not in the document]
Comment (alejandro-ripoll) [reply sdk_container/src/third_party/coreos-overlay/sys-kernel/bootengine/bootengine-9999.ebuild:1]: Done. Thanks

[the remark this answers is not in the document]
Comment (alejandro-ripoll) [reply sdk_container/src/third_party/coreos-overlay/sys-kernel/bootengine/bootengine-9999.ebuild:1]: Hi, the changelog entry should be added to bootengine repository as part of https://github.com/flatcar/bootengine/pull/96, right?
Thanks

**Comment by ader1990** [comment]
Hello, we need to be careful with this breaking change, as the OpenStack config drive is created *ONCE* when the instance is first provisioned and the metadata inside will never change afterwards, while the HTTP metadata is changing everytime the instance properties change.

For example, the metadata contains network interfaces information - in this case, the config drive ISO gets created with the network interface information from the initial instance create and will not change if a new network interface is added to the VM. 

Whereas the HTTP network metadata will change whenever such OpenStack instance (VM) network adapter is added / removed.

**Comment by ader1990** [comment]
@tormath1 This change gives precedence to the config drive, this is why it is a breaking change.

In most scenarios, it should not be an issue (other than the mentioned above, when during the lifecycle of the VM, the config drive contents diverge - are not updated vs the HTTP metadata).

Maybe we can get some input from OpenStack Flatcar users before we get this merge?

I think at this moment, the change is okay as long as it gets a proper changelog changes entry.

File: sdk_container/src/third_party/coreos-overlay/sys-kernel/bootengine/bootengine-9999.ebuild
[the remark this answers is not in the document]
Comment (ader1990) [reply sdk_container/src/third_party/coreos-overlay/sys-kernel/bootengine/bootengine-9999.ebuild:1]: Also, please squash the commits into one, with a clear commit message and description refering to the upstream change from the bootengine [and ](https://github.com/flatcar/bootengine/pull/96) https://github.com/coreos/afterburn/pull/462.

Usually, on a new release, users read the flatcar/scripts changelog and not the bootengine part.

**Comment by alejandro-ripoll** [comment]
Raised https://github.com/flatcar/scripts/pull/2010 including only changelog entry.
Thanks

**Comment by alejandro-ripoll** [comment]
@tormath1, sorry but I think I messed the PR up since I recreated my flatcar/scripts fork.
Now it won't let me reopen the PR since the repository that submitted the PR has been deleted.

**Comment by alejandro-ripoll** [comment]
> Hello @andrejro, CI is 🟢
> 
> Checking the logs, I see this:
> 
> ```
> May 27 18:27:49.026041 coreos-metadata[746]: May 27 18:27:49.025 WARN failed to locate config-drive, using the metadata service API instead
> May 27 18:27:49.066829 coreos-metadata[746]: May 27 18:27:49.066 INFO Fetching http://169.254.169.254/latest/meta-data/hostname: Attempt #1
> May 27 18:27:49.086081 coreos-metadata[746]: May 27 18:27:49.085 INFO Fetch successful
> May 27 18:27:49.087884 coreos-metadata[746]: May 27 18:27:49.087 INFO wrote hostname ci-9999-0-1-t-ab370a5df6.novalocal to /sysroot/etc/hostname
> ```
> 
> while the current behavior is:
> 
> ```
> May 28 05:18:13.018810 coreos-metadata[748]: May 28 05:18:13.018 INFO Fetching http://169.254.169.254/latest/meta-data/hostname: Attempt #1
> May 28 05:18:13.037701 coreos-metadata[748]: May 28 05:18:13.037 INFO Fetch successful
> May 28 05:18:13.040184 coreos-metadata[748]: May 28 05:18:13.040 INFO wrote hostname ci-3983-0-0-n-ebad85c034.novalocal to /sysroot/etc/hostname
> ```
> 
> so it looks good - the metadata service first tries to use config-drive then it fallbacks on the metadata service. Can I ask you to test in your case? The image is available here: http://bincache.flatcar-linux.net/images/amd64/9999.0.1+tormath1-pr-2009-openstack/flatcar_production_openstack_image.img
> 
> Thanks!

Tested this image on my environment and seems to be working as expected. Thanks
```
May 29 12:19:09 localhost coreos-metadata[886]: May 29 12:19:09.477 INFO wrote hostname REDACTED to /sysroot/etc/hostname
May 29 12:19:09 localhost systemd[1]: flatcar-openstack-hostname.service: Deactivated successfully.
May 29 12:19:09 localhost systemd[1]: Finished flatcar-openstack-hostname.service - Flatcar OpenStack Metadata Hostname Agent.
```

**Pull request closed without merging**